manim/mobject/three_dimensions.py

```diff
@@ -153,6 +156,81 @@ def set_fill_by_checkerboard(self, *colors, opacity=None):
+        ----------
+        axes :
+            The axes for the parametric surface.
+        colors :
+            A list of colors, ordered from lower z-values to higher z-values. If a list of tuples is passed
+            containing colors paired with numbers, then those numbers will be used as the pivots.
+
+        Returns
+        -------
+        :class:`~.ParametricSurface`
+            The parametric surface with a gradient applied by value
+
+        Examples
+        --------
+        .. manim:: Fill_by_valueExample
+            :save_last_frame:
+
+            class Fill_by_valueExample(ThreeDScene):
+                def construct(self):
+                    resolution_fa = 84
+                    self.set_camera_orientation(phi=75 * DEGREES, theta=-120 * DEGREES)
+                    axes = ThreeDAxes(x_range=(0, 5, 1), y_range=(0, 5, 1), z_range=(-1, 1, 0.5))
+                    def param_surface(u, v):
+                        x = u
+                        y = v
+                        z = np.sin(x) * np.cos(y)
+                        return z
+                    surface_plane = ParametricSurface(
+                        lambda u, v: axes.c2p(u, v, param_surface(u, v)),
+                        resolution=(resolution_fa, resolution_fa),
+                        v_min=0,
+                        v_max=5,
+                        u_min=0,
+                        u_max=5)
+                    surface_plane.set_style(fill_opacity=1)
+                    surface_plane.set_fill_by_value(axes=axes, colors=[(RED, -0.4), (YELLOW, 0), (GREEN, 0.4)])
+                    self.add(axes, surface_plane)
+        """
+        if type(colors[0]) is tuple:
+            new_colors, pivots = [[i for i, j in colors], [j for i, j in colors]]
+        else:
+            new_colors = colors
+
+            pivot_min = axes.z_range[0]
+            pivot_max = axes.z_range[1]
+            pivot_frequency = (pivot_max - pivot_min) / (len(new_colors) - 1)
+            pivots = np.arange(
+                start=pivot_min, stop=pivot_max + pivot_frequency, step=pivot_frequency
+            )
+
+        for mob in self.family_members_with_points():
+            z_value = axes.point_to_coords(mob.get_midpoint())[2]
+            if z_value <= pivots[0]:
+                mob.set_color(new_colors[0])
+            elif z_value >= pivots[-1]:
+                mob.set_color(new_colors[-1])
+            else:
+                for i, pivot in enumerate(pivots):
+                    if pivot > z_value:
+                        color_index = (z_value - pivots[i - 1]) / (
+                            pivots[i] - pivots[i - 1]
+                        )
+                        color_index = min(color_index, 1)
+                        mob_color = interpolate_color(
+                            new_colors[i - 1], new_colors[i], color_index
+                        )
+                        mob.set_color(mob_color, family=False)
+                        break
```

Comment:
**Nitpick (non-blocking)** : You can shorten this by using numpy. 
To get the closest matching value of z_value in `pivots`, you can do something like that `np.abs(pivots - z_value).argmin()`. This will give the index of the closest value. Beware, it can be the value _above_ z_value, so the line before won't directly work. Check [this](https://stackoverflow.com/questions/2566412/find-nearest-value-in-numpy-array) for the general idea. 

But it's a true nitpick; it's just a matter of code esthetic. I'm fine with how it is right now.

Reply:
If it is all the same I would rather keep it as is.  I know that this code works and it has been tested extensively.

Reply:
As you wish.